src-v5/new/carousel.tsx

```diff
@@ -8,18 +8,62 @@ import defaultProps from './default-carousel-props';
 const Carousel = (props: CarouselProps): React.ReactElement => {
   const [currentSlide, setCurrentSlide] = useState<number>(0);
   const [direction, setDirection] = useState<Directions | null>(null);
+  const [pause, setPause] = useState<boolean>(false);
+  const count = React.Children.count(props.children);
+  const timer: MutableRefObject<ReturnType<typeof setTimeout> | null> = useRef(
+    null
+  );
```

Comment:
```suggestion
  const timer = useRef<ReturnType<typeof setTimeout> | null>(null);
```

_Minor_: I think you could simplify this type slightly